client/src/components/project-selection/ProjectSelector.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from "react";
+import {
+  DeleteProject,
+  PostAddNewProject,
+  ProjectType,
+  GetChangeBetweenProjects,
+  PutEditProject,
+  getProjects,
+} from "../../api/api";
+import { FormControl, InputLabel, MenuItem, Select } from "@mui/material";
+import { useSnackbar } from "../../hooks/useSnackbar";
+import ProjectItem from "./ProjectItem";
+import NewProjectModal from "./NewProjectModal";
+import EditProjectModal from "./EditProjectModal";
+import NewProjectItem from "./NewProjectItem";
+const ProjectSelector = () => {
+  const [projects, setProjects] = useState<ProjectType[]>(getProjects());
+
+  const [newProjectModalIsOpen, setNewProjectModalIsOpen] = useState(false);
+  const [newProjectName, setNewProjectName] = useState("");
+
+  const [editProjectModalIsOpen, setEditProjectModalIsOpen] = useState(false);
+  const [editedProjectName, setEditedProjectName] = useState("");
+  const [selectedProjectName, setSelectedProjectName] = useState("");
+
+  const { show: showSnackbar, component: snackBar } = useSnackbar();
+
+  //remove from selected project edit and delete
+
+  //Change between projects
+  const handleChangeProject = (projectClick: any) => {
+    console.log("projectClick");
+    GetChangeBetweenProjects(projectClick); //api call
+    // setProjects((projects) => {
+    //   const updatedProjects = projects.map((project) => ({
+    //     ...project,
+    //     isOpen: project.name === projectClick.target.value,
+    //   }));
+
+    //   console.log(updatedProjects); // Log for checking
+    //   return updatedProjects;
+    // });
+  };
+
+  const handleDeleteProject = (e: React.MouseEvent, projectName: string) => {
+    e.stopPropagation();
+    DeleteProject(e, projectName); //api call
+
+    // console.log("delete project", projectName);
+    // setProjects((prevProjects) =>
+    //   prevProjects.filter((project) => project.name !== projectName)
+    // );
+  };
+
+  //Add new project functions
+  const handleNewProjectSave = () => {
+    // Add new project logic here
+    if (
```

Comment:
Separate the validations and provide different feedback to the user of each scenario.

Reply:
I didn't understand what you referring to